src/main/xerus/monstercat/api/Player.kt

```diff
@@ -160,10 +170,12 @@ object Player : FadingHBox(true, targetHeight = 25) {
 				seekBar.transitionToHeight(Settings.PLAYERSEEKBARHEIGHT(), 1.0)
 			}
 		}
+		getEqualizer()!!.enabledProperty().bind(Settings.ENABLEEQUALIZER)
```

Comment:
ist das nicht die Aufgabe des Equalizers, das im Listener zu binden?

Reply:
Das stimmt, aber für eine saubere Architektur sollte zuvor eine Aufteilung der Klassen in View (Präsentation), Model (Logik/Daten) und Controller (Steuerung) geschehen.

Die Settings sollten nicht als Hub statischer Methoden agieren, sondern dynamisch auf die Models verteilt werden. Man könnte bspw ein ApplicationModel machen mit den zentralen Einstellungen, welches ein EqualizerModel beherbergt (welches wiederum die EnableEqualizer-Einstellung beinhalten würde). Nun könnte man per Databinding/Observables/Listeners das EqualizerModel mit dem EqualizerView synchronisieren. Der Vorteil dieser Herangehensweise wäre eine Entkopplung von Programmlogik und Präsentation/UI.

Ob einem dieser Aufwand wert ist wäre die nächste Frage, nur grundsätzlich stimme dir da zu.

Reply:
Der hier referenzierte Equalizer bezieht sich auf den JavaFX-Equalizer im MediaPlayer, also auf das "player"-Attribut in "Player". Dementsprechend ist es eigentlich schon Aufgabe der Playerklasse hier das "View" mit den Settings zu verbinden.

Reply:
jap, mein Settings-Framework braucht dringend ein rework, ich weiß